Bryan/nestjs

blbergo opened the pull request:
# PR Checklist
- [x] Briefly describe your changes and include the issue number if it exists
- Migrate from Hono -> Nest.js
- Migrate from Drizzle -> TypeORM
- [x] Did you build your code? (`pnpm build`)
- [x] Does your code pass all tests (`pnpm test`)? Did you add new tests to for your changes? 
- Added new unit tests
- [x] Did you add any new environment variables? If so, did you notify an admin to update infisical?
- Added new env vars - will send out once the kms service is deployed.
- [ ] Did you document your changes? Do you need to update any of the `docs`?
- TODO

blbergo committed:
feat: nest setup

blbergo committed:
feat: nest setup

blbergo committed:
fix: use main instead of sub-app

blbergo committed:
fix: use main instead of sub-app

blbergo committed:
feat: db service

blbergo committed:
feat: db service

blbergo committed:
feat: kv service

blbergo committed:
feat: kv service

blbergo committed:
feat: queue service

blbergo committed:
feat: queue service

blbergo committed:
feat: keystone service

blbergo committed:
feat: keystone service

blbergo committed:
feat: barbican setup

blbergo committed:
feat: barbican setup

blbergo committed:
feat: full barbican service

blbergo committed:
feat: full barbican service

blbergo committed:
feat: swift service

blbergo committed:
feat: swift service

blbergo committed:
fix: refactor infra

blbergo committed:
fix: refactor infra

blbergo committed:
feat: auth service

blbergo committed:
feat: auth service

blbergo committed:
feat: otp service

blbergo committed:
feat: otp service

blbergo committed:
feat: sprint 1

blbergo committed:
feat: sprint 1

blbergo committed:
feat: sprint 1 (0 was prev)

blbergo committed:
feat: sprint 1 (0 was prev)

blbergo committed:
feat: sprint 2

blbergo committed:
feat: sprint 2

blbergo committed:
feat: sprint 3

blbergo committed:
feat: sprint 3

blbergo committed:
feat: login migration

blbergo committed:
feat: login migration

blbergo committed:
feat: improved refresh token logic

blbergo committed:
feat: improved refresh token logic

blbergo committed:
fix: proper refresh token expiration logic

blbergo committed:
fix: proper refresh token expiration logic

blbergo committed:
feat: password reset

blbergo committed:
feat: password reset

blbergo committed:
feat: mfa via totp

blbergo committed:
feat: mfa via totp

blbergo committed:
chore: refactor

blbergo committed:
chore: refactor

blbergo committed:
feat: video uploads

blbergo committed:
feat: video uploads

blbergo committed:
feat: download

blbergo committed:
feat: download

blbergo committed:
feat: remove unused specs and basic infra unit tests

blbergo committed:
feat: remove unused specs and basic infra unit tests

blbergo committed:
feat: swift integration tests

blbergo committed:
feat: swift integration tests

blbergo committed:
chore: cleanup

blbergo committed:
chore: cleanup

blbergo committed:
feat: otp unit tests

blbergo committed:
feat: otp unit tests

blbergo committed:
feat: auth unit tests

blbergo committed:
feat: auth unit tests

blbergo committed:
feat: typeorm + org schema migration

blbergo committed:
feat: typeorm + org schema migration

blbergo committed:
feat: org migration

blbergo committed:
feat: org migration

blbergo committed:
feat: unit + partial user setup

blbergo committed:
feat: unit + partial user setup

blbergo committed:
feat: complete user migrations

blbergo committed:
feat: complete user migrations

blbergo committed:
feat: video migrations

blbergo committed:
feat: video migrations

blbergo committed:
feat: assignment migration

blbergo committed:
feat: assignment migration

blbergo committed:
fix: update db tests

blbergo committed:
fix: update db tests

blbergo committed:
fix: auth service tests

blbergo committed:
fix: auth service tests

blbergo committed:
feat: otp tests

blbergo committed:
feat: otp tests

blbergo committed:
feat: final unit tests

blbergo committed:
feat: final unit tests

blbergo committed:
fix: linting

blbergo committed:
fix: linting

blbergo committed:
feat: auth service

blbergo committed:
feat: auth service

blbergo committed:
feat: video service

blbergo committed:
feat: video service

blbergo committed:
feat: migrate to redpanda

blbergo committed:
feat: migrate to redpanda

blbergo committed:
feat: otp tests

blbergo committed:
feat: otp tests

blbergo committed:
feat: auth service tests

blbergo committed:
feat: auth service tests

blbergo committed:
feat: video service tests

blbergo committed:
feat: video service tests

blbergo committed:
feat: auth controller e2e

blbergo committed:
feat: auth controller e2e

blbergo committed:
feat: partial video e2es

blbergo committed:
feat: partial video e2es

blbergo committed:
feat: linting and tests

blbergo committed:
feat: linting and tests

blbergo committed:
feat: jwt middleware tests

blbergo committed:
feat: jwt middleware tests

blbergo committed:
chore: kv + barbican

blbergo committed:
chore: kv + barbican

blbergo committed:
feat: s3 + keystone

blbergo committed:
feat: s3 + keystone

blbergo committed:
chore: swift

blbergo committed:
chore: swift

blbergo committed:
chore: swift

blbergo committed:
chore: swift

blbergo committed:
chore: queue service

blbergo committed:
chore: queue service

blbergo committed:
chore: docs

blbergo committed:
chore: docs

blbergo committed:
fix: config import and start script

blbergo committed:
fix: config import and start script

blbergo committed:
fix: migration command

blbergo committed:
fix: migration command

PattyCakesxD commented on apps/api/src/infra/db/entity/unit/file.ts:
```diff
@@ -0,0 +1,37 @@
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({
+    type: 'varchar',
+    length: 20,
+  })
+  type;
```
what does this col do/represent?

blbergo replied:
this is the file type - aka the extension. Not sure how relevant it will be, but this could either be the extension `.mp4` for example or the mime type `video/mp4` since we could also include the file extension in the path url as well.

ksarina256 committed:
Sarina 41 teamunit crud routes (#51)

* Added units schema and repository with CRUD + unit users

* added unitService.ts unitHandler.ts unitRoutes.http

* updated unitRoutes.http added unitService.ts simpleRateLimits.ts unitHandler.test.ts

* updated integration tests

* updated unit.ts combined w/unitRepository + added Team/Unit CRUD routes, invite flow, and user management

* added function comments + updated units.ts and index.ts

* fixed database constraints and user-unit relationships + added integration tests

* implement team/unit CRUD w/ rate limiting and improved type safety

* updated integration tests

* Remove coverage files from git and add to .gitignore

- Remove entire apps/api/coverage/ directory from version control
- Add coverage/ and *.lcov to .gitignore to prevent future commits
- Coverage reports should be generated locally/CI only, not committed

* updated lock file

* updated yaml files

* updated tests + units doc

* updated rate-limit.ts

* added delete unit implementation w/ tests

* updated response function + del comment

XmanJL committed:
Jasper 38 profile rud routes (#48)

* update .gitignore for macos

* fix macos launch

* feat: updated user schema

* feat(users routes): repository functions, routing logic, and schema validation

* test: validate users route endpoints

* fix: remove PII comments

* refactor(api): separate /users logic

* fix: separate users schema logic, remove status code in users schema, and more efficient impl for user query

* fix(users repository functions): use HTTP error helper instead

* refactor(users): added business logic

* refactor: remove unnecessary constructor

* feat: added functionality to upload/delete objects from the actual storage

* fix(http test): added X-User

* fix: merge fix

---------

Co-authored-by: Bryan <bryan.l.bergo@gmail.com>
Co-authored-by: Bryan Bergo <93395362+blbergo@users.noreply.github.com>

blbergo committed:
Merge remote-tracking branch 'refs/remotes/origin/bryan/nestjs' into bryan/nestjs

blbergo committed:
fix: cleanup remaining hono files

blbergo committed:
feat: profile migration

blbergo committed:
chore: refactor enums to config

blbergo committed:
feat: permission middleware

blbergo committed:
chore: tests

blbergo committed:
feat: soft delete user

blbergo committed:
feat: update profile

blbergo committed:
feat: GET users in unit

blbergo committed:
feat: invite system

blbergo committed:
feat: pfp uploads

blbergo committed:
chore: middleware tests

blbergo committed:
chore: me tests

blbergo committed:
chore: invite tests

blbergo committed:
chore: users unit tests

blbergo committed:
chore: me e2e

blbergo committed:
feat: users e2e tests

blbergo committed:
chore: final e2es

blbergo committed:
wip: ratelimiting

blbergo committed:
feat: ratelimitting

blbergo commented:
# Update
- integrated features from @ksarina256 and @XmanJL 
    - wrote unit + e2e tests for these feature
- created the permissions middleware + db tables, role:route map is stored in db + kv so we can update them without needing to build a new docker image
-  basic rate limitting, more is needed but for now this is ok.
